### DISCUSSION TITLE
Is panic on invalid format string normal?

### DISCUSSION THREAD

**Discussion by zhiburt** [time-rs/time#294] [Question] [answered]
This code will panic when I would expect it to be a error.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5e2d7cf7c26e660fdea8dc54e692576f

```rust
fn main() {
    time::Date::parse("", "%B %e, %Y");
}
```

```
thread 'main' panicked at 'Invalid specifier `e`', /playground/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.2.22/src/format/parse_items.rs:11:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by jhpratt** [comment, marked as answer]
Yes, this was a design choice made, as an invalid format description is almost certainly a programmer error. If you are using a user-inputted format description, you should validate it with `time::util::validate_format_string` to avoid panicking.

In time 0.3, the parsing of the format description and the formatting/parsing itself is split into two stages, such that an invalid format description can be handled independently.

**Reply by zhiburt** [reply]
Thank you for the answer and for pointing me to `time::util::validate_format_string`
